In [73]:
#Dependencies
import pandas as pd

In [74]:
#importing csv files to panda data frame
measurements_df= pd.read_csv("Resources\hawaii_measurements.csv")
stations_df= pd.read_csv("Resources\hawaii_stations.csv")

In [75]:
#checking overall info and inspect missing values of measurements
print(measurements_df.head())
print(measurements_df.info())
print(measurements_df.isnull().sum())

       station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
station    19550 non-null object
date       19550 non-null object
prcp       18103 non-null float64
tobs       19550 non-null int64
dtypes: float64(1), int64(1), object(2)
memory usage: 611.0+ KB
None
station       0
date          0
prcp       1447
tobs          0
dtype: int64


In [76]:
#checking overall info and inspect missing values of stations

print(stations_df.info())
print(stations_df.isnull().sum())
stations_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 5 columns):
station      9 non-null object
name         9 non-null object
latitude     9 non-null float64
longitude    9 non-null float64
elevation    9 non-null float64
dtypes: float64(3), object(2)
memory usage: 440.0+ bytes
None
station      0
name         0
latitude     0
longitude    0
elevation    0
dtype: int64


station                                    name  latitude  longitude  \
0  USC00519397                    WAIKIKI 717.2, HI US   21.2716  -157.8168   
1  USC00513117                    KANEOHE 838.1, HI US   21.4234  -157.8015   
2  USC00514830  KUALOA RANCH HEADQUARTERS 886.9, HI US   21.5213  -157.8374   
3  USC00517948                       PEARL CITY, HI US   21.3934  -157.9751   
4  USC00518838              UPPER WAHIAWA 874.3, HI US   21.4992  -158.0111   

   elevation  
0        3.0  
1       14.6  
2        7.0  
3       11.9  
4      306.6

The missing value in the measurement dataframe is 1447 which is less than 10%. Hence droping all the missing value will not have effect on the analysis.where as the stations data has no missing value.

In [77]:
#Droping missing values
clean_hawaii_measurements=measurements_df.dropna(how="any").reset_index(drop=True)
clean_hawaii_measurements.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18103 entries, 0 to 18102
Data columns (total 4 columns):
station    18103 non-null object
date       18103 non-null object
prcp       18103 non-null float64
tobs       18103 non-null int64
dtypes: float64(1), int64(1), object(2)
memory usage: 565.8+ KB


In [85]:
#for the purpose of creating primary key an empty id column is created in both csv files
#clean_hawaii_measurements.insert(0,'id','')
stations_df.insert(0,'id','')

In [83]:
clean_hawaii_measurements.head()

id      station        date  prcp  tobs
0     USC00519397  2010-01-01  0.08    65
1     USC00519397  2010-01-02  0.00    63
2     USC00519397  2010-01-03  0.00    74
3     USC00519397  2010-01-04  0.00    76
4     USC00519397  2010-01-07  0.06    70

In [89]:
#exporting the clean data as csv files
clean_hawaii_measurements.to_csv("Resources\clean_hawaii_measurements.csv",index=False)
stations_df.to_csv("Resources\clean_hawaii_stations.csv",index=False)